cpp/src/utilities/device_atomics.cuh

```diff
@@ -757,3 +781,215 @@ cudf::timestamp atomicCAS(cudf::timestamp* address, cudf::timestamp compare, cud
     return cudf::detail::typesAtomicCASImpl<T, sizeof(T)>()(address, compare, val);
 }
 
+
+/* Overloads for `atomicAnd` */
+/** -------------------------------------------------------------------------*
+ * @brief reads the `old` located at the `address` in global or shared memory, 
+ * computes (old & val), and stores the result back to memory at the same
+ * address. These three operations are performed in one atomic transaction.
+ *
+ * The supported types for `atomicAnd` are:
+ *   singed/unsigned integer 8/16/32/64 bits
+ * Cuda natively supports `sint32`, `uint32`, `sint64`, `uint64`.
+ *
+ * @param[in] address The address of old value in global or shared memory
+ * @param[in] val The value to be computed
+ *
+ * @returns The old value at `address`
+ * -------------------------------------------------------------------------**/
+__forceinline__ __device__
+int8_t atomicAnd(int8_t* address, int8_t val)
+{
+    return cudf::genericAtomicOperation(address, val, cudf::DeviceAnd{});
+}
+
+/**
+ * @overload uint8_t atomicAnd(uint8_t* address, uint8_t val)
+ */
+__forceinline__ __device__
+uint8_t atomicAnd(uint8_t* address, uint8_t val)
+{
+    return cudf::genericAtomicOperation(address, val, cudf::DeviceAnd{});
+}
+
+/**
+ * @overload int16_t atomicAnd(int16_t* address, int16_t val)
+ */
+__forceinline__ __device__
+int16_t atomicAnd(int16_t* address, int16_t val)
+{
+    return cudf::genericAtomicOperation(address, val, cudf::DeviceAnd{});
+}
+
+/**
+ * @overload uint16_t atomicAnd(uint16_t* address, uint16_t val)
+ */
+__forceinline__ __device__
+uint16_t atomicAnd(uint16_t* address, uint16_t val)
+{
+    return cudf::genericAtomicOperation(address, val, cudf::DeviceAnd{});
+}
+
+/**
+ * @overload int64_t atomicAnd(int64_t* address, int64_t val)
+ */
+__forceinline__ __device__
+int64_t atomicAnd(int64_t* address, int64_t val)
+{
+    using T = long long int;
```

Comment:
Why do we need overloads for int64_t and uint64_t if CUDA natively supports them as you say in the comment?

Reply:
nvcc compiler errors with without explicit int64_t, int64_t declaration.
It seems nvcc distincts int64_t and long long int.